tgui/packages/tgui/index.js

```diff
@@ -136,6 +137,20 @@ const setupApp = () => {
     store.dispatch(backendUpdate(state));
   };
 
+  window.reinit = newRef => {
+    logger.log("Reinitializing to: " + newRef, store.getState());
+    window.__ref__ = newRef;
+    callByond('', {
+      src: window.__ref__,
+      action: 'tgui:initialize',
+    });
+  };
```

Comment:
This is a source of a lag, because it becomes a long chain of byond calls.

```
window.reinit -> act('tgui:initialize') -> window.update
```

This could just be a single update callback:

```js
window.update = () => {
  ...
  if (state.standby) {
    logger.log("Reinitializing to: " + state.config.ref);
    window.__ref__ = state.config.ref;
  }
  ...
};
```

If window is recycled, just send a full proper update. I guarantee you will save a bunch of frames.